src/compiler/checker.ts

```diff
@@ -28781,7 +28781,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     spread = getSpreadType(spread, createJsxAttributesType(), attributes.symbol, objectFlags, /*readonly*/ false);
                     attributesTable = createSymbolTable();
                 }
-                const exprType = getReducedType(checkExpressionCached(attributeDecl.expression, checkMode));
+                const exprType = getReducedType(checkExpressionCached(attributeDecl.expression));

```

Comment:
this aligns the logic here for the equivalent call in the `checkObjectLiteral`:
https://github.dev/microsoft/TypeScript/blob/4ea8ec49e8aed6601fb6b978fd08031ec058da04/src/compiler/checker.ts#L28541

note that neither of those lines were touched recently